panos_upgrade_assurance/check_firewall.py

```diff
@@ -1194,3 +1201,173 @@ def run_snapshots(self, snapshots_config: Optional[List[Union[str, dict]]] = Non
+            2. The device is running software that is affected by the issue AND IS running user-id service OR data
+                redistribution
+
+        # Parameters
+
+        fail_when_affected_version_only (bool, optional): (defaults to `True`) When set to False, this test will only
+            fail if the software version is affected by the root certificate issue, AND the device is used for data
+            redistribution OR it's using an out-of-date content DB version.
+        """
+        result = CheckResult()
+
+        software_version = self._node.get_device_software_version()
+
+        # note; '-h' is substituted out of these versions to keep with semantic versioning
+        fixed_version_map = {
+            "81": [("==", "8.1.21.2"), (">=", "8.1.25.1")],
+            "90": [(">=", "9.0.16.5")],
+            "91": [
+                ("==", "9.1.11.4"),
+                ("==", "9.1.12.6"),
+                ("==", "9.1.13.4"),
+                ("==", "9.1.14.7"),
+                ("==", "9.1.16.3"),
+                (">=", "9.1.17"),
+            ],
+            "100": [
+                ("==", "10.0.8.10"),
+                ("==", "10.0.11.3"),
+                (">=", "10.0.12.3"),
+            ],
+            "101": [
+                ("==", "10.1.3.2"),
+                ("==", "10.1.5.3"),
+                ("==", "10.1.6.7"),
+                ("==", "10.1.8.6"),
+                ("==", "10.1.9.3"),
+                (">=", "10.1.10"),
+            ],
+            "102": [
+                ("==", "10.2.3.9"),
+                (">=", "10.2.4"),
+            ],
+            "110": [
+                ("==", "11.0.0.1"),
+                ("==", "11.0.1.2"),
+                (">=", "11.0.2"),
+            ],
+            "111": [
+                (">=", "11.1.0"),
+            ],
+        }
+        fixed_content_version = 8776.8390
+
+        fixed_versions = fixed_version_map.get(f"{software_version.major}{software_version.minor}")
+        if fixed_versions:
+            for operator, fixed_version in fixed_versions:
+                fixed_version = parse_version(fixed_version)
+                if operator == "==":
+                    if software_version == fixed_version:
+                        result.status = CheckStatus.SUCCESS
+                elif operator == ">=":
+                    if software_version >= fixed_version:
+                        result.status = CheckStatus.SUCCESS
+
+        # If the device is already running fixed software, we can return immediately
+        if result.status == CheckStatus.SUCCESS:
+            return result
+
+        # Return if this check is just looking at the software and not implementing any other checks
+        if fail_when_affected_version_only:
+            result.status = CheckStatus.FAIL
+            result.reason = "Device is running a software version that is impacted by the device root certificate" "expiry."
```

Comment:
String splitted into to 2.
```suggestion
            result.reason = "Device is running a software version that is impacted by the device root certificate expiry."
```